src/server/frontend_wayland/wl_surface.h

```diff
@@ -85,7 +102,8 @@ class WlSurface : public wayland::Surface
     geometry::Size buffer_size() const { return buffer_size_; }
     bool synchronized() const;
 
-    void set_role(WlSurfaceRole* role_);
+    void instill_role(WlSurfaceRole* role_);
+    void annul_role();
```

Comment:
Hmm... Not terribly happy with these names. Maybe `set_role()` and `remove_role()`?